what-is-software-development/study-like-a-programmer.md

```diff
@@ -0,0 +1,340 @@
+
+#### What You Know
+
+First, let's convince ourselves that there are different levels of knowledge and learning. Knowing and understanding something has more nuance than "I get it" or "I don't get it."
+
+Here's a common model of "knowing things" that we can reference:
+
+Level | Depth of knowledge | Description
+--- | --- | ---
+1 | I know what I know | I know what I've been taught
+2 | I don't know what I don't know | I'm learning enough to know there's more to learn
+3 | I know what I don't know | I've learned enough to know what I've learned vs. what more to learn
+4 | I don't know what I know | I don't even realize how much I've learned
+
+Levels 1 and 2 should describe where everyone is right now with programming. In fact, we should spend most of our times as learners in level 1 and 2.
+
+Level 3 is the destination that we want all of our careers to point to.
+
+Level 4 describes an experienced programmer. It is _not_ our destination.
+
+**We understand that it takes practice and time to figure out "what you know" vs. "what you don't know."** Follow below to find practical ways to help you figure that out.
+
+#### How to Ask Questions
+
+The primary way to level up in knowledge is to get _better_ at asking questions and seek answers. **How well you can solve a problem is directly related to how practiced you are at asking questions.**
+
+**The kinds of questions that you can form will help you figure out what answers you need.** Here is one flowchart that can help you.
+
+Level of Understanding | What kinds of Questions You Have | Action
+--- | --- | ---
+Knowledge | What does ... mean? What is the standard syntax for ...? Where can I find the definition for ...? | Use flashcards for specific vocabulary, concepts, and syntax
+Understanding | What are some examples about ...? When is ... relevant and not relevant? What are variations of this syntax? | [Create a crib sheet/cheat sheet](https://en.wikipedia.org/wiki/Cheat_sheet). Try explaining the concept to someone else
+Application | Can I create examples of ...? Can I repeat and modify examples of ...? If someone says that this concept is relevant, can you find a way to apply this concept and fit it into the bigger picture? | Take time to create your own working examples, and check them. Make diagrams, skits, songs, etc. and share them.
+Analysis | On a bigger scale of systems... What factors impact this? What consequences are there? What are the technical trade-offs? | Research relevant examples. Compare and contrast similar examples. Make tables and lists to compare.
+Evaluate | On a bigger scale of systems... When should I use, defend, or critique this? | Research. Hypothesize, then make short experiments. Read various sources.
+Create | On a bigger scale of systems... How do I orchestrate this with a system of other concepts? Can I create something that uses this? | Research. Planning. Hypothesizing, then make short experiements. Read various sources.
```

Comment:
```suggestion
Level of Understanding | What kinds of Questions You Have | Action
--- | --- | ---
Knowledge | What does ... mean? What is the standard syntax for ...? Where can I find the definition for ...? | Use flashcards for specific vocabulary, concepts, and syntax
Understanding | What are some examples about ...? When is ... relevant and not relevant? What are variations of this syntax? | [Create a crib sheet/cheat sheet](https://en.wikipedia.org/wiki/Cheat_sheet). Try explaining the concept to someone else
Application | Can I create examples of ...? Can I repeat and modify examples of ...? If someone says that this concept is relevant, can you find a way to apply this concept and fit it into the bigger picture? | Take time to create your own working examples, and check them. Make diagrams, skits, songs, etc. and share them.
Analysis | On a bigger scale of systems... What factors impact this? What consequences are there? What are the technical trade-offs? | Research relevant examples. Compare and contrast similar examples. Make tables and lists to compare.
Evaluate | On a bigger scale of systems... When should I use, defend, or critique this? | Research. Hypothesize, then make short experiments. Read various sources.
Create | On a bigger scale of systems... How do I orchestrate this with a system of other concepts? Can I create something that uses this? | Research. Planning. Hypothesizing, then make short experiements. Read various sources.
Destruct | What would make ____ break? Would using _____ break another part of the code? | Break your code. Read about errors and what makes them appear.
```

I LOVE this chart. I added a 'Destruct' row as I think it's valuable to teach students to learn how to 'break' things in order to be more comfortable with errors.

Reply:
This is awesome. I'm gonna think on this; contextually, this is supposed to borrow from a teaching/learning framework. I wonder if there's space for this in the "how to break down a problem" resource?